packages/@aws-c2a/cdk-pipelines-step/lib/private/change-analysis-check.ts

```diff
@@ -0,0 +1,173 @@
+        },
+      },
+      buildSpec: codebuild.BuildSpec.fromObject({
+        version: 0.2,
+        phases: {
+          build: {
+            commands: [
+              'npm install -g aws-c2a',
+              // $CODEBUILD_INITIATOR will always be Code Pipeline and in the form of:
+              // "codepipeline/example-pipeline-name-Xxx"
+              'export PIPELINE_NAME="$(node -pe \'`${process.env.CODEBUILD_INITIATOR}`.split("/")[1]\')"',
+              'payload="$(node -pe \'JSON.stringify({ "PipelineName": process.env.PIPELINE_NAME, "StageName": process.env.STAGE_NAME, "ActionName": process.env.ACTION_NAME })\' )"',
+              // ARN: "arn:aws:codebuild:$region:$account_id:build/$project_name:$project_execution_id$"
+              'ARN=$CODEBUILD_BUILD_ARN',
+              'REGION="$(node -pe \'`${process.env.ARN}`.split(":")[3]\')"',
+              'ACCOUNT_ID="$(node -pe \'`${process.env.ARN}`.split(":")[4]\')"',
+              'PROJECT_NAME="$(node -pe \'`${process.env.ARN}`.split(":")[5].split("/")[1]\')"',
+              'PROJECT_ID="$(node -pe \'`${process.env.ARN}`.split(":")[6]\')"',
+              'export PIPELINE_LINK="https://$REGION.console.aws.amazon.com/codesuite/codepipeline/pipelines/$PIPELINE_NAME/view?region=$REGION"',
+              // Run invoke only if cdk diff passes (returns exit code 0)
+              // 0 -> true, 1 -> false
+              ifElse({
+                condition: `aws-c2a diff --app "assembly-${props.codePipeline.stack.stackName}-$STAGE_NAME/" --broadening-permissions --fail`,
+                thenStatements: [
+                  invokeLambda,
+                  'export MESSAGE="No changes that violate rules detected."',
+                ],
+                elseStatements: [
+                  'aws-c2a html --report report.json',
+                  putObject,
+                  `export LINK=$(${signObject})`,
+                  `[ -z "\${NOTIFICATION_ARN}" ] || ${publishNotification}`,
+                  'export MESSAGE="Deployment would make security-impacting changes. Click the link below to inspect them, then click Approve if all changes are expected."',
+                ],
+              }),
+            ],
+          },
+        },
+        env: {
+          'exported-variables': [
+            'LINK',
+            'MESSAGE',
+          ],
+        },
+      }),
+    });
+
+    // this is needed to check the status the stacks when doing `cdk diff`
+    this.c2aDiffProject.addToRolePolicy(new iam.PolicyStatement({
+      actions: ['sts:AssumeRole'],
+      resources: ['*'],
+      conditions: {
+        'ForAnyValue:StringEquals': {
+          'iam:ResourceTag/aws-cdk:bootstrap-role': ['deploy'],
+        },
+      },
+    }));
+
+    this.c2aDiffProject.addToRolePolicy(new iam.PolicyStatement({
+      actions: ['cloudformation:GetTemplate', 'cloudformation:DescribeStackResources', 'cloudformation:DescribeStacks'],
+      resources: ['*'],
+    }));
```

Comment:
For some reason I had to do this in order to let the codebuild perform the GetTemplate and what not.. I'm guessing this is not ideal, but unsure how we can narrow the scope of the resources glob

Reply:
This is fine for now I think.